VMEncryption/main/DiskUtil.py

```diff
@@ -397,6 +456,47 @@ def luks_get_uuid(self, header_or_dev_path):
                 return splits[1]
         return None
 
+    def get_token_id(self,header_or_dev_path,token_name):
+        '''if LUKS2 header has token name return the id else return none.'''
+        if not header_or_dev_path or not os.path.exists(header_or_dev_path) or not token_name:
+            self.logger.log("get_token_id: invalid input, header_or_dev_path:{0} token_name:{1}".format(header_or_dev_path,token_name))
+            return None
+        luks_dump_out = self._luks_get_header_dump(header_or_dev_path)
+        tokens = self._extract_luksv2_token(luks_dump_out)
+        for token in tokens:
+            if token[1] == token_name:
+                return token[0]
+        return None
+
+    def restore_luks2_token(self, device_name=None):
+        '''this function restoring token type Azure_Disk_Encryption_BackUp to Azure_Disk_Encryption, 
+        this function acts on 4 secenarios. [token id Azure_Disk_Encryption, token id Azure_Disk_Encryption_BackUp, resote_action)'''
+        '''[Y,Y,remove token type Azure_Disk_Encryption_BackUp]
+           [Y,N,do nothing]
+           [N,N,do nothing]
+           [N,Y,move token type Azure_Disk_Encryption_BackUp to Azure_Disk_Encryption]'''
```

Comment:
For each of the scenarios i.e [Y,Y],[Y,N],[N,N],[N,Y] , could you also please add the documentation of the scenarios which causes each of these states.